### ISSUE TITLE
Week 11 data

### ISSUE THREAD

**Issue by dlangenfeld** [UM-R-for-EnvSci-Registered-Student/General-Discussion#21]
Are we supposed to be using the data from the npp_dry column? when I try to plot this data it comes out really weird

**Comment by peperg** [comment]
@dlangenfeld,

Yes, that is the right variable.

Quick hint, NPP cannot be -9999, yet there are a bunch of those... Why? What do we do with them?

**Comment by dlangenfeld** [comment]
@peperg I'm not exactly sure what -9.9999 means, is it supposed to be 0? Are we supposed to use string replace?

**Comment by madelinestanley** [comment]
@dlangenfeld 
Sometimes -9999 is assumed as N.A.or missing.

In Week 10, Pepe mentioned that sometimes people use half of the detection limit, which is where you can use the string function to remove the > symbol to then extract and divide those values. But in this case, they don't provide the detection limit and this value is technically N.A./missing, so what I did was use the function filter to extract those rows from my analysis. I think that is the correct thing to do. Hopefully @peperg  can confirm on this before I give any bad advice

**Comment by dlangenfeld** [comment]
@madelinestanley thank you! I have heard that before, but the value has a decimal so I'm not sure if that changes the meaning at all? That definitely makes the most sense to just not include those values

**Comment by madelinestanley** [comment]
@dlangenfeld 
It may be marked as -9.999e+04, so still -9999. 
I hope that helps, and also hope that's the correct  process. Hopefully @peperg  will be able to doubly confirm with you!

**Comment by peperg** [comment]
Exactly, the "-9999" here is used to indicate an NA. You can see how such a use is a very bad idea... It is a number... so it doesn't get flagged as a problem! if you are not looking for it, it can end up in your analysis. In this case you noticed because you plotted it and the actual values of the NPP are quite different from -9999, but if we would have gone ahead and model without looking, we would have never known and the analysis might have been published that way. Also, note how this is easier to miss the more data you have, and specially if you have large datasets that combine data from multiple sources, which is the way the field is going. So in summary... don't do it! 

But in terms of the code, The approach that @madelinestanley mentions of filtering works. An even easier way, is to take advantage of the options that `read_csv()` already gives you.

Because this kind of nonsensical way of reporting NAs is common place, `read_csv()` has an option to indicate a list of possible strings in the file which will be considered NAs. Take a look at the function info [here](https://readr.tidyverse.org/reference/read_delim.html). I am talking about the `na = ` part. 

The default is `na = c("", "NA")` which means that it will read any empty space or any "NA" and an NA. You can explicitly call this when you load your data using `reac_csv()` and specify that your dataset has one of these nonsensical cases. For example"

``` R

rea_csv("the_data.csv", na = c("", "NA", "-9999"))

```

**Comment by dlangenfeld** [comment]
@peperg that is so helpful thank you! So I tried doing this but it didn't work, is this the proper placement? 

kelp_data <- tibble(files = fs::dir_ls(here("data"))) %>%  
  mutate(data = pmap(list(files), 
                     ~ read_csv(..1, col_names = TRUE, na = c("", "NA", "-9999"))))

**Comment by peperg** [comment]
@dlangenfeld ,

Yes, that is the proper placement, the problem is, and i am sorry i miss-typed it earlier, that the NAs are coded as "-99999". Five nines instead of 4!

Pepe

**Comment by leahdickenson** [comment]
@dlangenfeld @madelinestanley @peperg 

Thank you all for commenting about this! I didn't see this until after I completed the assignment so I went back and removed the -99999 from the read_csv, and it made a huge difference to the results/figures!!
